core/src/config.rs

```diff
@@ -28,82 +27,120 @@ impl<'de> Deserialize<'de> for MacAddr {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
     where
         D: Deserializer<'de>,
     {
-        let s = <String>::deserialize(deserializer)?;
+        let s = String::deserialize(deserializer)?;
         Ipv4Cidr::from_str(&s).map_err(de::Error::custom)
     }
 }
 
 // make `Ipv6Cidr` serde deserializable.
 impl<'de> Deserialize<'de> for Ipv6Cidr {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
     where
         D: Deserializer<'de>,
     {
-        let s = <String>::deserialize(deserializer)?;
+        let s = String::deserialize(deserializer)?;
         Ipv6Cidr::from_str(&s).map_err(de::Error::custom)
     }
 }
 
-/// Runtime settings.
+/// Deserializes a duration from seconds expressed as `u64`.
+pub fn duration_from_secs<'de, D>(deserializer: D) -> std::result::Result<Duration, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let secs = u64::deserialize(deserializer)?;
+    Ok(Duration::from_secs(secs))
+}
+
+/// Deserializes an option of duration from seconds expressed as `u64`.
+pub fn duration_option_from_secs<'de, D>(
+    deserializer: D,
+) -> std::result::Result<Option<Duration>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    // for now this is the cleanest way to deserialize an option, till a better
+    // way is implemented, https://github.com/serde-rs/serde/issues/723
+    #[derive(Deserialize)]
+    struct Wrapper(#[serde(deserialize_with = "duration_from_secs")] Duration);
+
+    let option = Option::deserialize(deserializer)?.and_then(|Wrapper(dur)| {
+        if dur.as_secs() > 0 {
+            Some(dur)
+        } else {
+            None
+        }
+    });
+    Ok(option)
+}
+
+/// Runtime config settings.
 #[derive(Clone, Deserialize)]
-pub struct RuntimeSettings {
+#[serde(deny_unknown_fields)]
+pub struct RuntimeConfig {
     /// Application name. This must be unique if you want to run multiple
     /// DPDK applications on the same system.
     pub app_name: String,
 
     /// Indicating whether the process is a secondary process. Secondary
     /// process cannot initialize shared memory, but can attach to pre-
     /// initialized shared memory by the primary process and create objects
-    /// in it. The default value is `false`.
+    /// in it. Defaults to `false`.
+    #[serde(default)]
     pub secondary: bool,
 
     /// Application group name. Use this to group primary and secondary
     /// processes together in a multi-process setup; and allow them to share
     /// the same memory regions. The default value is the `app_name`. Each
     /// process works independently.
+    #[serde(default)]
     pub app_group: Option<String>,
 
     /// The identifier of the master core. This is the core the main thread
-    /// will run on. The default value is `0`.
+    /// will run on.
     pub master_core: CoreId,
```

Comment:
so, we want to force users to have to put `master_core` in their config (and not rely on defaults)?

Reply:
yeah not clear what's better, so err on the side of being more explicit.